src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Runtime/UseStringEqualsOverStringCompare.Fixer.cs

```diff
@@ -0,0 +1,218 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Immutable;
+using System.Composition;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Analyzer.Utilities;
+using Analyzer.Utilities.Extensions;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.Editing;
+using Microsoft.CodeAnalysis.Operations;
+
+using Resx = Microsoft.NetCore.Analyzers.MicrosoftNetCoreAnalyzersResources;
+using RequiredSymbols = Microsoft.NetCore.Analyzers.Runtime.UseStringEqualsOverStringCompare.RequiredSymbols;
+
+namespace Microsoft.NetCore.Analyzers.Runtime
+{
+    [ExportCodeFixProvider(LanguageNames.CSharp, LanguageNames.VisualBasic), Shared]
+    public sealed class UseStringEqualsOverStringCompareFixer : CodeFixProvider
+    {
+        public override async Task RegisterCodeFixesAsync(CodeFixContext context)
+        {
+            var document = context.Document;
+            var token = context.CancellationToken;
+            var semanticModel = await document.GetSemanticModelAsync(token).ConfigureAwait(false);
+
+            if (!RequiredSymbols.TryGetSymbols(semanticModel.Compilation, out var symbols))
+                return;
+
+            var root = await document.GetSyntaxRootAsync(token).ConfigureAwait(false);
+
+            if (semanticModel.GetOperation(root.FindNode(context.Span, getInnermostNodeForTie: true), token) is not IBinaryOperation violation)
+                return;
+
+            //  Get the replacer that applies to the reported violation. Bail out if no replacer 
+            //  matches the reported violation.
+            var replacer = GetOperationReplacers(symbols).FirstOrDefault(x => x.IsMatch(violation));
+            if (replacer is null)
+                return;
+
+            var codeAction = CodeAction.Create(
+                Resx.UseStringEqualsOverStringCompareCodeFixTitle,
+                CreateChangedDocument,
+                nameof(Resx.UseStringEqualsOverStringCompareCodeFixTitle));
+            context.RegisterCodeFix(codeAction, context.Diagnostics);
+            return;
```

Comment:
return is not needed here

Reply:
@ryzngard It's a convention used in the repository (and dotnet/roslyn as well) to put an explicit return just before local functions

Reply:
o.o who decided that? Because I definitely haven't been

Reply:
Apparently this is @CyrusNajmabadi preferrence... I won't block a PR on style preference that isn't documented. You can keep if you like, and I'll just grumble in my corner alone :P

Reply:
I've added https://github.com/dotnet/roslyn-analyzers/issues/5122 for further discussion

Reply:
@mavasani has asked me to have a `return` before local the local function definitions for clarity.

Reply:
Yea, I followed up offline with him and a few others. I was unaware that was a practice we started. Hopefully we can get an analyzer to help the underlying issue, but for now this is correct.